src/stateManager/StateManager.ts

```diff
@@ -349,6 +370,20 @@ class StateManager {
             await this.stateMachine.stateTransition(transaction);
         const encodedState = await this.stateMachine.getState();
 
+        //This is done before the state snapshot is created
+        //This is because the exit channels need to be taken into account when creating the state snapshot
+        this.handleExitChannelsStorage(
+            exitChannels,
+            Number(transaction.header.forkCnt),
+            Number(transaction.header.transactionCnt)
+        );
+
+        this.handleStateSnapshotStorage(
```

Comment:
same stuff - fork and height are not relevant to store exitChannels and you need to wrap them in an ExitChannelBlock (it points to the previous block and includes all exitChannels that are returned from the STF).

Where [fork][height] are useful here are for 'markers' so we know where to create a milestone if the participant set contracted if (stateMachine.getParticipants().length < same thing before STF) 

Plus you called the handle twice

Reply:
ok for the first part will change, and the handle twice i called handleExit channel and the handleStateSnapshot to avoid having boilerplate

Reply:
Ok I see the handler wraps them in the block - good

Reply:
i have updated the code